### PR TITLE
Stories/1208 hover intent on menu

### PR DESCRIPTION
**Pull request by Sjardo**
Fixes https://github.com/Yoast/yoast.com/issues/1208

For testing, when hovering over a main-menu item, it gives a small delay when it switches to it's active/hover state. After a new-submenu shows, you can move quickly over all other menu-items to the submenu bar and back.

### CODE REVIEW

File: js/src/nav-menu.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ jQuery( document ).ready( function( $ ) {
 	 * Sets jQuery event listeners to improve the visual flow of 'current menu item parents'.
 	 */
 	function init_active_item_flow() {
-		var parent_class = 'current-menu-parent';
+		var parent_class = 'current-menu-parent hover';
```
Comment (diedexx) [inline js/src/nav-menu.js]: This adds the hover class to all parent items. I think you'd want to remove the hover class here and let hover_intent add the hover class

File: js/src/nav-menu.js [hunk 2]
```diff
@@ -61,6 +61,30 @@ jQuery( document ).ready( function( $ ) {
...
+    /**
+     * Checks the intention of hovering on a main-menu item or just moving over to reach a submenu item.
+     */
+    function init_hover_intent() {
+        jQuery("ul.yoast-main-menu").hoverIntent({
+            over: yoast_show_submenu,
+            out: yoast_hide_submenu,
+            selector: '> li',
+            timeout: '250'
+
```
Comment (diedexx) [inline js/src/nav-menu.js]: Please remove this newline

File: css/sass/3-particles/_yoast-main-menu.scss [hunk 3]
```diff
@@ -126,7 +126,7 @@
       }
     }
 
-    li:hover, li:focus {
+    li.hover, li:focus{
```
Comment (diedexx) [inline css/sass/3-particles/_yoast-main-menu.scss]: Please fix a small codestyle issue: add the space before the `{` again.

File: footer-wp-activate.php [hunk 4]
```diff
@@ -0,0 +1,4 @@
+<?php
```
Comment (diedexx) [inline footer-wp-activate.php]: What are the footer-\*, header-\* and activate-\* files for?

**Comment by atimmer** [comment]
We really should not add this much in one go. Let's explore CSS3 pure solutions: https://www.google.nl/search?client=opera&q=css+hoverintent&sourceid=opera&ie=UTF-8&oe=UTF-8

File: footer-wp-activate.php (hunk 4 above)
Comment (Sjardo) [inline footer-wp-activate.php]: After changing the pull-request to new-yoast-com, you will see your comment no longer applies

**Comment by Sjardo** [comment]
@atimmer I have check a css3 solution, but wont fix the problem. CSS3 wont "read" the intention of the user, as the :hover will no longer apply of you move over an other main-menu item. Therefore the JS solution.

File: js/src/nav-menu.js (hunk 1 above)
Comment (Sjardo) [inline js/src/nav-menu.js]: Replaced setting the 'hover' class to 'current-menu-partent' from line 30 to the 'init_hover_intent' function

**Comment by Sjardo** [comment]
We have green light from Joost to implement the hoverintent, if the code is correct.

**Comment by NicolleY** [comment]
Acceptance 🚧, 3 side-effect should be solved:

1) Hovering both "SEO-blog" and "Plugins" menu items doesn't work well. In some cases it doesn't show the corresponding submenu items and color in the row below the main menu items. The behavior is inconsistent, but happens for sure while hovering those menu items from outside the menu.
2) The row with submenu items disappears shortly, when hovering several menu items or hovering these items from outside the menu. 
3) Optional: the intended hover effects should be a little bit faster, it feels a bit slow now while hovering over the menu items.

Furthermore, the intended hover effect works fine.

**Comment by Sjardo** [comment]
After replacing way i add the class "hover" to the current menu item, the issues seem to be solved

**Comment by NicolleY** [comment]
CR done 👍

**Comment by maartenleenders** [comment]
Acceptance 🚧 

Perhaps we should try delays that are shorter than 200ms, as that is quit long. Anything between 80 and 100 ms seems to work just as well.

Also, moving back and forth quickly between the labels can cause the "current label" to refocus. This leads to a flickering effect.

(And finally (but unrelated to your PR, there seems to be a 4 4 0 px padding on the list items, which means not the entire list item is a clickable link. This seems to be unnecessary.)

**Comment by Sjardo** [comment]
After discussion with Tim, Omar and Dushanthi, we close this issue due to the fact we will remove the submenu in the near future.

**Pull request closed without merging**